### PR TITLE
Quick review of healthchecker library

### PR DESCRIPTION
**Pull request by keadwen**
Updates:
  * Remove HealthChecker interface and functions (unnecessary boilerplate)
  * Refactor the naming in structs (avoid Getters, Setters and New)
  * Make Run a receiver of HealthChecker struct
  * Enable context passing to terminate all tests

### CODE REVIEW

File: healthchecker/healthchecker.go [hunk 1]
```diff
@@ -1,57 +1,60 @@
...
+
+func (hc HealthCheck) Run(ctx context.Context) {
+	go func() {
+		for {
+			select {
+			case <-time.After(hc.Interval):
+				hc.runSingleCheck()
+				fmt.Printf("Name: %q\tHealth: %v\n", hc.Name, hc.IsHealthy())
+			case <-ctx.Done():
+				fmt.Printf("Name: %q\tContext terminated", hc.Name)
```
Comment (jbszczepaniak) [inline healthchecker/healthchecker.go:40]: shouldn't this case return?

File: main.go [hunk 2]
```diff
@@ -9,34 +10,35 @@ import (
...
+
+	redisCheck := h.HealthCheck{
+		Name: "get some key from redis",
+		FailedThreshold: 1,
+		PassedThreshold: 1,
+		Interval: 3*time.Second,
+		S: h.NewRedisBased("some_key"),
+	}
+
+	fileCheck.Run(ctx)
```
Comment (jbszczepaniak) [inline main.go:39]: Run() as a part of heatlhchecker just looks better :+1:

File: healthchecker/healthchecker.go (hunk 1 above)
Comment (keadwen) [reply healthchecker/healthchecker.go:40]: That's right. I would include an error return in the function signature & change the line to be:
case <- ctx.Done:
  [...]whatever print
  return fmt.Errorf("Context terminated") // or ctx.Err()

**Comment by keadwen** [comment]
PTAL

**Pull request merged**